### PR TITLE
Removing deprecated destruct/remember syntax _eqn.

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** removal of deprecated syntax

The `_eqn` syntax was deprecated since 2012, so I guess there is no scrupuls to have. Don't know if the change log is really worth.

- [X] Entry added in the changelog

### CODE REVIEW

File: doc/changelog/04-tactics/11877-master+deprecated-_eqn.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Removed:**
+  Deprecated syntax `_eqn` for :tacn:`destruct` and :tacn:`remember`
```
Comment (Zimmi48) [inline doc/changelog/04-tactics/11877-master+deprecated-_eqn.rst]:
```suggestion
  Deprecated syntax `_eqn` for :tacn:`destruct` and :tacn:`remember`.
  Use `eqn:` syntax instead
```

File: plugins/ltac/g_tactic.mlg [hunk 2]
```diff
@@ -461,10 +461,6 @@ GRAMMAR EXTEND Gram
   ;
   eqn_ipat:
     [ [ IDENT "eqn"; ":"; pat = naming_intropattern -> { Some (CAst.make ~loc pat) }
-      | IDENT "_eqn"; ":"; pat = naming_intropattern ->
-         { warn_deprecated_eqn_syntax ~loc "H"; Some (CAst.make ~loc pat) }
```
Comment (Zimmi48) [inline plugins/ltac/g_tactic.mlg]: You probably should remove the `warn_deprecated_eqn_syntax` as well.

**Comment by Zimmi48** [comment]
Hi @herbelin! What about my suggestion to the changelog?

**Comment by herbelin** [comment]
Sorry, I had missed it. It is welcomed and I committed it.

**Review by Zimmi48 — APPROVED** [review]
The fix has already been tested in CI before the changelog update. In the aim of minimizing computational resources, I'm merging this without relaunching the pipeline.

**Pull request merged**